store/registry.go

```diff
@@ -140,9 +146,56 @@ func (rs *registryStore) FindAgentGlobally(agentID string) (*config.AgentConfig,
 	if err != nil {
 		return nil, fmt.Errorf("failed to get the latest ref: %v, agentID: %s", err, agentID)
 	}
+
 	return loadBot(rs.ctx, rs.cfg, rs.mc, agentID, agt.Manifest)
 }
 
+func (rs *registryStore) FindScannerShardIDForBot(agentID, scannerAddress string) (uint, error) {
+	chainID := rs.cfg.ChainID
+
+	// get manifest cid
+	agt, err := rs.FindAgentGlobally(agentID)
+	if err != nil {
+		return 0, err
+	}
+
+	// fetch manifest
+	agentManifest, err := rs.mc.GetAgentManifest(rs.ctx, agt.Manifest)
+	if err != nil {
+		return 0, err
+	}
+
+	// extract chain setting from manifest
+	chainSetting, ok := agentManifest.Manifest.ChainSettings[uint(chainID)]
```

Comment:
I could just also check for the ["default"] entry, which I think means this map needs to be string-keyed.